### ISSUE TITLE
Broken become_pass function with delegate_to other host in inventory

### ISSUE THREAD

**Issue by 4ndyZ** [mitogen-hq/mitogen#1005]
* Which version of Ansible are you running?
Ansible 5.10.0 (also happens on Ansible 6.7.0)

* Is your version of Ansible patched in any way?
No

* Are you running with any custom modules, or `module_utils` loaded?
No

* Have you tried the latest master version from Git?
Yes

* Do you have some idea of what the underlying problem may be?
  Since the update to version `0.3.4` I am no longer able to use `delegate_to` in combination with `become_method: su` to another host in the inventory because the su become method fails on the delegated host.

  It looks like the changes in commit f1503874de82353cbed8b51408d20fdfa899f8f7 (see: https://github.com/mitogen-hq/mitogen/pull/1003/commits/f1503874de82353cbed8b51408d20fdfa899f8f7#diff-a98bb8f749289d5b203cc23fbe75d1a1e651bf8b20a80d4e3a7598129f1a4472R439-R442) introduced a new way to optain the `become_pass`. This changes breaks `delegate_to` for me.

  When reverting the changes and using the following code with the old way and using the `_play_context`everything works as expected.
  ```python
         def become_pass(self):
              return optional_secret(self._play_context.become_pass)
     ```

   The following playbook can be used to verify the problem.
    ```yml
    hosts: all
    tasks:
      - name: Template on other host using variables from current host
        delegate_to: other_host
        block:
          - name: Template config_1 on other host using variables from current host
            ansible.builtin.template:
              src: template_1.conf.j2
              dest: "/etc/somedir/config_1.conf"
              owner: user
              group: user
              mode: "0600"
      
          - name: Template config_2 on other host using variables from current host
            ansible.builtin.template:
              src: template_2.conf.j2
              dest: "/etc/somedir/config_2.conf"
              owner: user
              group: user
              mode: "0755"

* Mention your host and target OS and versions
   Host system: macOS 13.4.1
   Target system:
   ```shell
   cat /etc/redhat-release
   CentOS Linux release 7.9.2009 (Core)
   ```

* Mention your host and target Python versions
   Host system:
   ```shell
   python3 --version
   Python 3.11.4
   ```
   Target system:
   ```shell
   python3 --version
   Python 3.6.8
   ```

 * If reporting any kind of problem with Ansible, please include the Ansible version along with output of "ansible-config dump --only-changed".
   ```
    ansible-config dump --only-changed

    COLLECTIONS_PATHS(ansible/ansible.cfg) = ['ansible/collections']
    DEFAULT_BECOME(ansible/ansible.cfg) = True
    DEFAULT_BECOME_METHOD(ansible/ansible.cfg) = su
    DEFAULT_BECOME_USER(ansible/ansible.cfg) = root
    DEFAULT_HOST_LIST(ansible/ansible.cfg) = ['ansible/inventories/dev']
    DEFAULT_ROLES_PATH(ansible/ansible.cfg) = ['ansible/roles']
    DEFAULT_STDOUT_CALLBACK(ansible/ansible.cfg) = yaml
    DEFAULT_STRATEGY(ansible/ansible.cfg) = mitogen_linear
    DEFAULT_STRATEGY_PLUGIN_PATH(ansible/ansible.cfg) = ['ansible/lib/python3/site-packages/ansible_mitogen']
    DEFAULT_VAULT_PASSWORD_FILE(ansible/ansible.cfg) = ansible/scripts/open_vault.sh
    DIFF_ALWAYS(ansible/ansible.cfg) = True
    DIFF_CONTEXT(ansible/ansible.cfg) = 3
    HOST_KEY_CHECKING(ansible/ansible.cfg) = False
    RETRY_FILES_ENABLED(ansible/ansible.cfg) = False
   ```

**Comment by 4ndyZ** [comment]
Problem is solved with #1007 with reintroduction of the `playcontext` in https://github.com/mitogen-hq/mitogen/pull/1007/commits/07ae7daac001d8ede830ae246e1b86e1ebdcb8e0#diff-a98bb8f749289d5b203cc23fbe75d1a1e651bf8b20a80d4e3a7598129f1a4472R447

Project still needs a new release tag to introduce the bugfix to a release version.

**Comment by 4ndyZ** [comment]
@moreati I hate to ask such questions, but when do plan to tag the changes introduced in the current master?

I really would like to bump my Ansible version to a more up to date version, but mitogen is a must have and therefor a show stopper for me :/
